### PR TITLE
Remove `assets:precompile` task enhancement

### LINKED ISSUE #125: Don't enhance precompile with yarn install
Running `yarn:install` during `asset:precompile` might be bit too magical. In the same way as we don't run `bundle install`, we shouldn't run `yarn install`.

`yarn:install` being ran adds some time to the compile. On Heroku also, by default it will reinstall all packages even if they have been previously cached. Not good.

Prepping of Shakapacker 7 might be good time to get rid of this behaviour:

TODO:
- Remove `yarn:install` and `webpacker:yarn_install` tasks
- Don't enhance  `assets:precompile` tasks with yarn install (ref https://github.com/shakacode/shakapacker/blob/master/lib/tasks/yarn.rake#L36-L38 and https://github.com/shakacode/shakapacker/blob/master/lib/tasks/webpacker/compile.rake#L12)
- Provide instructions for re-implementing enhanced tasks if users want to keep the old behaviour (pretty much requires user to create a task like https://github.com/shakacode/shakapacker/blob/master/lib/tasks/yarn.rake)

### PR DESCRIPTION
**Pull request by jherdman**
Move this responsibility to the host application

Resolves #125

### CODE REVIEW

**Comment by jherdman** [comment]
Updated as requested.

File: lib/tasks/webpacker/compile.rake [hunk 1]
```diff
@@ -34,9 +17,5 @@ namespace :webpacker do
 end
 
 if Webpacker.config.webpacker_precompile?
-  if Rake::Task.task_defined?("assets:precompile")
-    enhance_assets_precompile
-  else
-    Rake::Task.define_task("assets:precompile" => ["webpacker:yarn_install", "webpacker:compile"])
-  end
+  Rake::Task.define_task("assets:precompile" => ["webpacker:compile"])
```
Comment (tomdracz) [inline lib/tasks/webpacker/compile.rake:20]: We should still use `enhance_assets_precompile` here so we need something like:

```
  if Rake::Task.task_defined?("assets:precompile")
    enhance_assets_precompile
  else
    Rake::Task.define_task("assets:precompile" => ["webpacker:compile"])
  end
```

**Review by tomdracz — CHANGES_REQUESTED** [review]
Two more changes and we're good to go @jherdman

**Comment by jherdman** [comment]
Fixed. Apologies for the bumps along the way on this one.

**Comment by justin808** [comment]
LGTM

**Review by justin808 — APPROVED** [review]

**Comment by tomdracz** [comment]
> I think we can remove this too? https://github.com/shakacode/shakapacker/blob/master/lib/install/bin/yarn

This is still used during setup and diagnostic check so good to remain. All looks good now, merging! Thanks @jherdman

**Comment by Judahmeek** [comment]
@tomdracz I don't understand the point of this PR.

Isn't `yarn install` is a no-op if `node_modules` is already up to date?

**Pull request merged**